src/include/planner/update_plan.h

```diff
@@ -79,7 +79,8 @@ class UpdatePlan : public AbstractPlan {
  private:
   // Initialize private members and construct colum_ids given a UpdateStatement.
   void BuildInitialUpdatePlan(const parser::UpdateStatement *parse_tree,
-                              std::vector<oid_t> &columns);
+                              std::vector<oid_t> &columns,
+                              concurrency::Transaction *txn = nullptr);
```

Comment:
This function want to pass the transaction to addTableWithName(), it transaction is null, addTableWIthName will begin a new transaction.